plugin/al/plugin/AL_USDMayaTestPlugin/py/CMakeLists.txt

```diff
@@ -1,5 +1,47 @@
+set(TEST_NAME TestUSDMayaPython)
+
+set(path 
+    "$ENV{PATH}"
+)
+
+set(pythonPath 
+    "${CMAKE_INSTALL_PREFIX}/plugin/al/lib/python"
+    "$ENV{PYTHONPATH}"
+)
+
+set(mayaPluginPath 
+    "${CMAKE_INSTALL_PREFIX}/plugin/al/plugin"
+)
+
+if(IS_WINDOWS)
+    string(REPLACE ";" "\;" pythonPath "${pythonPath}")
+    string(REPLACE ";" "\;" path "${path}")
+    string(REPLACE ";" "\;" mayaPluginPath "${mayaPluginPath}")
+else()
+    separate_arguments(pythonPath NATIVE_COMMAND "${pythonPath}")
+    separate_arguments(path NATIVE_COMMAND "${path}")
+    separate_arguments(mayaPluginPath NATIVE_COMMAND "${mayaPluginPath}")
+
+    string(REPLACE "\;" ":" pythonPath "${pythonPath}")
+    string(REPLACE "\;" ":" path "${path}")
+    string(REPLACE "\;" ":" mayaPluginPath "${mayaPluginPath}")
+endif()
+
 add_test(
-    NAME TestUSDMayaPython
+    NAME ${TEST_NAME}
     WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
-    COMMAND testUSDMayaPython.sh ${AL_INSTALL_PREFIX} ${USD_LIBRARY_DIR} ${CMAKE_CURRENT_SOURCE_DIR}
-    )
+    COMMAND ${MAYA_PY_EXECUTABLE} -c "from unittest import main;import maya.standalone; \
+                                      maya.standalone.initialize(name='python'); \
+                                      import testProxyShape;main(module=testProxyShape); \
+                                      import testTranslators;main(module=testTranslators); \
+                                      import testLayerManager;main(module=testLayerManager); \
+                                      maya.standalone.uninitialize();"
```

Comment:
There is a bug here that we just discovered. Only testProxyShape seems to run!!! Something to look at.

Reply:
Also - if my hunch is right (and unittest.main is calling sys.exit), then the maya.standalone.unitialize isn't being called. In which case, it should either be removed, or put inside a try/finally

Reply:
> This is pretty big - and I think I see why. unittest.main will automatically call sys.exit, I think. Probably we should just make each of these it's own test.

@elrond79 Good point! That was my plan too but if we want to do this, then each of these tests needs to have it's own unique names.  Is this something Animal Logic Ok with?

e.g for demonstration purposes ( these can be wrapped in a for loop )
```
add_test(
    NAME testProxyShape
    WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
    COMMAND ${MAYA_PY_EXECUTABLE} -c "from unittest import main;import maya.standalone; \
                                      maya.standalone.initialize(name='python'); \
                                      import testProxyShape;main(module=testProxyShape); \
                                      maya.standalone.uninitialize();"
)
add_test(
    NAME testTranslators
    WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
    COMMAND ${MAYA_PY_EXECUTABLE} -c "from unittest import main;import maya.standalone; \
                                      maya.standalone.initialize(name='python'); \
                                      import testTranslators;main(module=testProxyShape); \
                                      maya.standalone.uninitialize();"
)
add_test(
    NAME testLayerManager
    WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
    COMMAND ${MAYA_PY_EXECUTABLE} -c "from unittest import main;import maya.standalone; \
                                      maya.standalone.initialize(name='python'); \
                                      import testLayerManager;main(module=testProxyShape); \
                                      maya.standalone.uninitialize();"
)
```

Reply:
Well, I obviously can't answer for AL, but I don't see why they would have a problem with it.  Getting more granularity in your returned results is usually a good thing.  I suppose you're now initializing maya 4 times, instead of once, but hopefully everyone runs these tests in parallel, so that shouldn't be so bad.

I would still like to know if the uninitialize call is being run, though, and either axe it or try/finalize it...

Reply:
@elrond79  See 65f683934d964fb

I looked into this in more details today:

1- As you mentioned already,  `unittest.main` does in fact call `sys.exit`. After reading the docs, it turns out that this behaviour can be disabled by simply passing `exit=False`.

https://docs.python.org/2/library/unittest.html#unittest.main

2- `__name__ == __main__` block no longer is going to get called for these tests since there are now imported. I didn't want to remove `__name__ == __main__` block  from these tests but that's something that can be mentioned as a comment so it is clear.

3- Another issue that I noticed was that testTranslators.py failed but ctest still reported that the test passed.
We do have this issue in number of places and my solution would be call the `sys.exit` based the `TestProgram result`.

Quote from doc:
> Calling main actually returns an instance of the TestProgram class. This stores the result of the tests run as the 
result attribute.

4- I have disabled testTranslators.py since there are test failures based on some of the member variables not getting incremented in the ` importObject` method.

```
self.assertEqual(CubeGenerator.getState()["importObjectCount"],1)
5: AssertionError: 0 != 1

self.assertEqual(DeleteParentNodeOnPostImport.nbImport, 1)
5: AssertionError: 0 != 1
```

Reply:
Hmm... well, I think I'd still prefer making separate tests... or, barring that, bundling the tests from the three modules into a TestSuite, and then calling unittest.main once.

However, I think your approach should work, and at this point, I don't want to let the perfect be the enemy of the good.

I'm a little worried about the need to disable testTranslators.py, though... that failure might potentially be important.  Maybe as a compromise we can open a new PR re-enabling this test, and marking it a WIP, as a kind of placeholder to come back to this?

Reply:
> bundling the tests from the three modules into a TestSuite

I totally agree with this but that means more changes and for now it is something that we would like to avoid.

> I'm a little worried about the need to disable testTranslators.py, though... that failure might potentially be important. Maybe as a compromise we can open a new PR re-enabling this test, and marking it a WIP, as a kind of placeholder to come back to this?

Let's see what AL recommend here but I agree that this needs to be enabled. Creating a separate PR and enabling it is a good idea.

Reply:
Ok, I made a separate PR to help make sure we get this resolved:

https://github.com/Autodesk/maya-usd/pull/212

Reply:
@elrond79 That's great!